### PR TITLE
Vulkan 1.3.274 and SPIR-V SDK-1.3.268.0 updates

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by pthariensflame** [comment]
Seems to be ready!

File: Formula/s/spirv-cross.rb [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+  sha256 "dd656a51ba4c229c1a0bb220b7470723e8fd4b68abb7f2cf2ca4027df824f4a0"
   license all_of: [
     "Apache-2.0",
     "MIT",
     "CC-BY-4.0",
     :cannot_represent, # LicenseRef-KhronosFreeUse
   ]
   version_scheme 1
+  head "https://github.com/KhronosGroup/SPIRV-Cross.git",
+       branch: "main"
```
Comment (chenrui333) [inline Formula/s/spirv-cross.rb:14]:
```suggestion
  head "https://github.com/KhronosGroup/SPIRV-Cross.git", branch: "main"
```

Comment (chenrui333) [reply Formula/s/spirv-cross.rb:14]: why not move into the same line?

**Comment by chenrui333** [comment]
Great work! Thanks @pthariensflame!

File: Formula/v/vulkan-extensionlayer.rb [hunk 2]
```diff
@@ -56,15 +57,16 @@ def caveats
   test do
     ENV.prepend_path "VK_LAYER_PATH", share/"vulkan/explicit_layer.d"
     ENV["VK_ICD_FILENAMES"] = Formula["vulkan-tools"].lib/"mock_icd/VkICD_mock_icd.json"
+    ENV[OS.mac? ? "DYLD_LIBRARY_PATH" : "LD_LIBRARY_PATH"] =
+      Formula["vulkan-loader"].lib.to_s
```
Comment (carlocab) [inline Formula/v/vulkan-extensionlayer.rb:61]: Needing this looks suspicious. Seems like we should add this path to the binary rpaths instead?

File: Formula/v/vulkan-tools.rb [hunk 3]
```diff
@@ -103,6 +104,8 @@ def caveats
 
   test do
     ENV["VK_ICD_FILENAMES"] = lib/"mock_icd/VkICD_mock_icd.json"
+    ENV[OS.mac? ? "DYLD_LIBRARY_PATH" : "LD_LIBRARY_PATH"] =
+      Formula["vulkan-loader"].lib.to_s
```
Comment (carlocab) [inline Formula/v/vulkan-tools.rb:108]: Same here.

File: Formula/v/vulkan-validationlayers.rb [hunk 4]
```diff
@@ -96,13 +97,13 @@ def caveats
   test do
     ENV.prepend_path "VK_LAYER_PATH", share/"vulkan/explicit_layer.d"
     ENV["VK_ICD_FILENAMES"] = Formula["vulkan-tools"].lib/"mock_icd/VkICD_mock_icd.json"
+    ENV[OS.mac? ? "DYLD_LIBRARY_PATH" : "LD_LIBRARY_PATH"] =
+      Formula["vulkan-loader"].lib.to_s
```
Comment (carlocab) [inline Formula/v/vulkan-validationlayers.rb:101]: And here.

File: Formula/v/vulkan-volk.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        }
+      }
+    EOS
+    system ENV.cc, testpath/"test.c",
+           "-I#{include}", "-L#{lib}",
+           "-I#{Formula["vulkan-headers"].include}",
+           "-lvolk", "-D#{volk_static_defines}",
+           "-o", testpath/"test"
+    ENV[OS.mac? ? "DYLD_LIBRARY_PATH" : "LD_LIBRARY_PATH"] =
+      Formula["vulkan-loader"].lib.to_s
```
Comment (carlocab) [inline Formula/v/vulkan-volk.rb:64]: Ditto.

File: Formula/v/vulkan-extensionlayer.rb (hunk 2 above)
Comment (pthariensflame) [reply Formula/v/vulkan-extensionlayer.rb:61]: @carlocab How would we do that?  I'm not familiar enough with that side of things.

**Comment by pthariensflame** [comment]
@carlocab Ping?  I'm not sure how to proceed.

File: Formula/v/vulkan-extensionlayer.rb (hunk 2 above)
Comment (carlocab) [reply Formula/v/vulkan-extensionlayer.rb:60]: Try passing
```
-DCMAKE_INSTALL_RPATH=#{rpath(target: Formula["vulkan-loader"].opt_lib)}
```
to the first CMake invocation. Hopefully that will just work. If not we can try to tweak the `source:` argument of `#rpath`. If that doesn't work either we'll need to take a closer look at the CMake scripts.

File: Formula/s/spirv-tools.rb [hunk 6]
```diff
@@ -1,11 +1,16 @@
 class SpirvTools < Formula
   desc "API and commands for processing SPIR-V modules"
   homepage "https://github.com/KhronosGroup/SPIRV-Tools"
-  # TODO: Change `-DPYTHON_EXECUTABLE` to `-DPython3_EXECUTABLE` in next release.
-  # Ref: https://github.com/KhronosGroup/SPIRV-Tools/commit/a63ac9f73d29cd27cdb6e3388d98d1d934e512bb
-  url "https://github.com/KhronosGroup/SPIRV-Tools/archive/refs/tags/v2023.2.tar.gz"
-  sha256 "7416cc8a98a10c32bacc36a39930b0c5b2a484963df5d68f388ed7ffee1faad3"
+  url "https://github.com/KhronosGroup/SPIRV-Tools/archive/refs/tags/vulkan-sdk-1.3.268.0.tar.gz"
+  sha256 "4c19fdcffb5fe8ef8dc93d7a65ae78b64edc7a5688893ee381c57f70be77deaf"
```
Comment (chenrui333) [inline Formula/s/spirv-tools.rb:5]: looks like the changelog is using 2023.xx? https://github.com/KhronosGroup/SPIRV-Tools/blame/main/CHANGES

Comment (pthariensflame) [reply Formula/s/spirv-tools.rb:5]: Seems like the answer is the one I happened to pick; good for us!

**Review by chenrui333 — APPROVED** [review]
LGTM

**Review by p-linnane — APPROVED** [review]

**Comment by p-linnane** [comment]
> @chenrui333 Seems like this didn't get picked up?

PR's that contain new formula have to be manually merged by maintainers. Thanks for your patience as well as all your work here!

**Pull request merged**